File: js/problem-2.js
function fib(limit) {
  const arr = [1, 2];
  while (arr.length < limit) {
    const last = arr[arr.length - 1];
    const lastButOne = arr[arr.length - 2];
    arr.push(last + lastButOne);
  }
  return arr;
}

const result = fib(4000000);

console.log(result[result.length - 1]);

File: ruby/problem-2.rb
class Challenge
  attr_reader :sum

  def initialize
    @sum = 0
  end

  def fib(limit, cur, last)
    if cur >= limit then
      return last
    end
    if cur % 2 == 0 then
      @sum += cur
    end
    return fib(limit, cur + last, cur)
  end
end

c = Challenge.new
c.fib(4_000_000, 2, 1)

puts c.sum
